chainer/functions/loss/ctc.py

```diff
@@ -323,18 +328,15 @@ def connectionist_temporal_classification(
         raise TypeError('x must be a list of Variables')
     if not isinstance(blank_symbol, int):
         raise TypeError('blank_symbol must be non-negative integer.')
-    assert blank_symbol >= 0
-    assert blank_symbol < x[0].shape[1]
+    assert 0 <= blank_symbol < x[0].shape[1]
     # This implementation only supports 1-dimensional data.
     # TODO(jnishi): Support d(>1)-dimentinal inputs.
-    assert(len(x[0].shape) == 2)
+    assert x[0].ndim == 2
 
     if input_length is None:
-        xp = cuda.get_array_module(x[0].data)
-        input_length = variable.Variable(
-            xp.full((len(x[0].data),), len(x), dtype=numpy.int32))
-        label_length = variable.Variable(
-            xp.full((len(t.data),), len(t.data[0]), dtype=numpy.int32))
+        xp = cuda.get_array_module(x[0])
+        input_length = xp.full(x[0].shape[0], len(x), dtype=numpy.int32)
+        label_length = xp.full(t.shape[0], t.shape[1], dtype=numpy.int32)
```

Comment:
How about `t.shape[:2]`?

Reply:
No, its shape is `t.shape[0]` and its value is `t.shape[1]`.

Reply:
Oh, I see. Please use `len(t)`.